torch/csrc/jit/custom_operator.h

```diff
@@ -0,0 +1,190 @@
+#pragma once
+
+#include <torch/csrc/jit/function_schema.h>
+#include <torch/csrc/jit/ir.h>
+#include <torch/csrc/jit/operator.h>
+#include <torch/csrc/jit/stack.h>
+#include <torch/csrc/utils/variadic.h>
+
+#include <caffe2/utils/Metaprogramming.h>
+#include <caffe2/utils/TypeList.h>
+
+namespace torch { namespace jit {
+namespace detail {
+template <typename... Ts, size_t... Is>
+std::vector<Argument> createArgumentVectorFromTypes(Indices<Is...> indices) {
+  // Arguments are named "_<index>"
+  return {Argument("_" + std::to_string(Is), getTypePtr<Ts>())...};
+}
+
+template <typename... Ts, size_t... Is>
+std::vector<Argument> createReturns(Indices<Is...> indices) {
+  return createArgumentVectorFromTypes<Ts..., Is...>();
+}
+
+/// Unpack a tuple return type into a vector of return types, one per tuple
+/// element.
+template <typename... Ts>
+std::vector<Argument> createReturns(std::tuple<Ts...>* tuple) {
+  // Create an index pack so we can call `get<Indices>` on the tuple next.
+  return createReturns<Ts...>(typename MakeIndices<sizeof...(Ts)>::indices{});
```

Comment:
You can use the `guts::index_sequence` which is naming-wise closer to the standard.

Reply:
We would have to change that all over the codebase then